src/mavsdk/core/socket_holder.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#if defined(WINDOWS)
+#include <winsock2.h>
+#endif
+
+namespace mavsdk {
+
+class SocketHolder {
+public:
+    SocketHolder() noexcept = default;
+    explicit SocketHolder(int socket_fd) noexcept;
+
+    SocketHolder(SocketHolder&&) noexcept = default;
+    SocketHolder& operator=(SocketHolder&&) noexcept = default;
+
+    ~SocketHolder() noexcept;
+
+    void reset(int fd) noexcept;
+    void close() noexcept;
+
+    bool empty() const noexcept;
+    int get() const noexcept;
+
+#if defined(WINDOWS)
+    static constexpr int invalid_socket_fd = INVALID_SOCKET;
```

Comment:
Presumably to fix the Windows warning? :shrug: 

```suggestion
    static constexpr int invalid_socket_fd = static_cast<int>(INVALID_SOCKET);
```